op-e2e/e2eutils/setup.go

```diff
@@ -0,0 +1,228 @@
+package e2eutils
+
+import (
+	"math/big"
+	"os"
+	"path"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+)
+
+var testingJWTSecret = [32]byte{123}
+
+// WriteDefaultJWT writes a testing JWT to the temporary directory of the test and returns the path to the JWT file.
+func WriteDefaultJWT(t TestingBase) string {
+	// Sadly the geth node config cannot load JWT secret from memory, it has to be a file
+	jwtPath := path.Join(t.TempDir(), "jwt_secret")
+	if err := os.WriteFile(jwtPath, []byte(hexutil.Encode(testingJWTSecret[:])), 0600); err != nil {
+		t.Fatalf("failed to prepare jwt file for geth: %v", err)
+	}
+	return jwtPath
+}
+
+func uint642big(in uint64) *hexutil.Big {
+	return (*hexutil.Big)(new(big.Int).SetUint64(in))
+}
+
+// DeployParams bundles the deployment parameters to generate further testing inputs with.
+type DeployParams struct {
+	DeployConfig   *genesis.DeployConfig
+	MnemonicConfig *MnemonicConfig
+	Secrets        *Secrets
+	Addresses      *Addresses
+}
+
+// TestParams parametrizes the most essential rollup configuration parameters
+type TestParams struct {
+	MaxSequencerDrift   uint64
+	SequencerWindowSize uint64
+	ChannelTimeout      uint64
+}
+
+func MakeDeployParams(t require.TestingT, tp *TestParams) *DeployParams {
+	mnemonicCfg := DefaultMnemonicConfig
+	secrets, err := mnemonicCfg.Secrets()
+	require.NoError(t, err)
+	addresses := secrets.Addresses()
+	deployConfig := &genesis.DeployConfig{
+		L1ChainID:   901,
+		L2ChainID:   902,
+		L2BlockTime: 2,
+
+		MaxSequencerDrift:      tp.MaxSequencerDrift,
+		SequencerWindowSize:    tp.SequencerWindowSize,
+		ChannelTimeout:         tp.ChannelTimeout,
+		P2PSequencerAddress:    addresses.SequencerP2P,
+		OptimismL2FeeRecipient: common.Address{0: 0x42, 19: 0xf0}, // tbd
+		BatchInboxAddress:      common.Address{0: 0x42, 19: 0xff}, // tbd
+		BatchSenderAddress:     addresses.Batcher,
+
+		L2OutputOracleSubmissionInterval: 6,
+		L2OutputOracleStartingTimestamp:  -1,
+		L2OutputOracleProposer:           addresses.Proposer,
+		L2OutputOracleOwner:              common.Address{}, // tbd
+
+		L1BlockTime:                 15,
+		L1GenesisBlockNonce:         0,
+		CliqueSignerAddress:         common.Address{}, // proof of stake, no clique
```

Comment:
Is it worth it to just delete `CliqueSignerAddress`?

Reply:
That's part of the chainops, and other tests like hive may depend on that. We can remove it later, it's currently an optional field. And I like being explicit about not using clique, instead of omitting the field altogether for the default zero definition.